### PR TITLE
Embed attachments' text data for known file types

### PR DESCRIPTION
**Pull request by mocobeta**
#126 

Embeds the attachment text data if the file extension is known and the file size is not too large.

I use [&lt;details&gt; tag](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) (markup?) here since this is convenient when you want to collapse long texts by default.

e.g.: please click `▶` to expand the collapsed lines.
- patch file: https://github.com/mocobeta/migration-test-3/issues/569#issuecomment-1207398605
- java file: https://github.com/mocobeta/migration-test-3/issues/570

(note) "common file extensions" are detected by this script.

```python
from operator import itemgetter
from pathlib import Path
from collections import defaultdict

ATT_DIR = "/home/moco/tmp/lucene-jira-archive/attachments/"

path = Path(ATT_DIR)
extensions = defaultdict(int)

for f in path.glob("**/*"):
    if f.is_file():
        p = f.name.rsplit(".")
        if len(p) < 2:
            continue
        ext = p[1]
        extensions[ext] += 1

for (ext, count) in sorted(extensions.items(), key=itemgetter(1), reverse=True):
    if count > 2:
        print(f"{ext} ({count})")
```

### CODE REVIEW

File: migration/src/jira_util.py [hunk 1]
```diff
@@ -236,6 +237,31 @@ def extract_pull_requests(o: dict) -> list[str]:
 REGEX_GITHUB_ISSUE_LINK = re.compile(r"(\s)(#\d+)(\s)")
 
 
+# common file extensions in Lucene Jira attachments
+FILE_EXT_TO_LANG = {
```
Comment (mocobeta) [inline migration/src/jira_util.py:241]: Those extensions appear at least three times in all attachments.

Comment (vlsi) [reply migration/src/jira_util.py:241]: Thanks for commenting on GitHub, however, the comment would be valuable as a regular code comment 😉

**Comment by vlsi** [comment]
I would expand by default (there's details-summary expanded by default) the code blocks less than 10KiB or so.

File: migration/src/jira_util.py (hunk 1 above)
Comment (mocobeta) [reply migration/src/jira_util.py:241]: Ah yes, will do.

**Comment by mocobeta** [comment]
> I would expand by default (there's details-summary expanded by default) the code blocks less than 10KiB or so.

+1. I randomly picked source code files in my computer and decided the threshold for "open by default" to 5k bytes (100 lines or so).
https://github.com/apache/lucene-jira-archive/pull/136/commits/47b46328b1c08f0b84303d6bd39138d7363e22b2

**Comment by mocobeta** [comment]
@mikemccand I'll merge this tomorrow if there is no further feedback/suggestion. Let me know if we need some more time.

**Comment by mikemccand** [comment]
Wow, these inlined attachments for known extensions look beautiful!  +1 to merge.

But I noticed [comments like this one](https://github.com/mocobeta/migration-test-3/issues/570#issuecomment-1207401771) that seem to have failed to even turn text like `LUCENE-NNNN` into hyperlinks.  Is this just because you didn't run all phases of the migration on these few test issues?

**Comment by mocobeta** [comment]
> But I noticed [comments like this one](https://github.com/mocobeta/migration-test-3/issues/570#issuecomment-1207401771) that seem to have failed to even turn text like `LUCENE-NNNN` into hyperlinks. Is this just because you didn't run all phases of the migration on these few test issues?

Yes. To confirm this patch, we don't need to apply issue id remapping.

**Review by mikemccand — APPROVED** [review]

**Pull request merged**